### PR TITLE
feat: Audit trpc calls

### LINKED ISSUE #275: Go through all the TRPC calls and add missing Audit.create
Pretty much all the events that happens on the backend should be recorded on `Audit` table. Let's go through all the backend TRPC calls and make sure we are doing it.

### PR DESCRIPTION
**Pull request by VK-RED**
resolves #275

### CODE REVIEW

File: src/server/audit/schema.ts [hunk 1]
```diff
@@ -51,6 +55,37 @@ export const AuditSchema = z.object({
...
+    "role.created",
+    "role.updated",
+    "role.deleted",
+
+    "template.field-created",
+    "template.created",
+
+    "passkey.created",
+    "passkey.authOptions-created",
+    "passkey.registrationOptions-created",
```
Comment (dahal) [inline src/server/audit/schema.ts:86]: Please change both `authOptions... and registration...` to `passkey.updated`, details will be on summary

File: src/trpc/routers/company-router/router.ts [hunk 2]
```diff
@@ -63,6 +64,21 @@ export const companyRouter = createTRPCRouter({
             lastAccessed: new Date(),
           },
         });
+        const { user } = session;
+        await Audit.create(
+          {
+            action: "company.lastAccessed",
```
Comment (dahal) [inline src/trpc/routers/company-router/router.ts:70]: I don't think we need this, let's remove it.

File: src/server/audit/schema.ts [hunk 3]
```diff
@@ -5,8 +5,12 @@ export const AuditSchema = z.object({
   action: z.enum([
     "user.signed-up",
     "user.onboarded",
+    "user.verified",
+    "user.password-updated",
 
     "company.created",
+    "company.lastAccessed",
```
Comment (dahal) [inline src/server/audit/schema.ts:12]: I don't think we need this, lets remove `lastAccessed` audit

File: src/server/audit/schema.ts [hunk 4]
```diff
@@ -51,6 +55,37 @@ export const AuditSchema = z.object({
 
     "update.public-status",
     "update.private-status",
```
Comment (dahal) [inline src/server/audit/schema.ts:57]: While you are on this, can you please change these two status with `update.updated`, details will be on summary

**Review by dahal — CHANGES_REQUESTED** [review]
Great work on this PR @VK-RED , it looks good overall . Please make these recommended changes.

**Comment by VK-RED** [comment]
Will update it soon @dahal !!

**Review by dahal — APPROVED** [review]

**Pull request merged**